.github/workflows/upgrade-tests.yml

```diff
@@ -0,0 +1,223 @@
+name: Upgrade Tests
+
+on:
+  pull_request:
+    paths:
+      # Server-side packages that affect upgrade compatibility
+      - "packages/worker/**"
+      - "packages/server/**"
+      - "packages/backend-core/**"
+      - "packages/shared-core/**"
+      # The upgrade tests themselves
+      - "packages/upgrade-tests/**"
+      # Docker and hosting changes that affect the build
+      - "hosting/**"
```

Comment:
We could also add the root `package.json` or the `yarn.lock` (even if this might cause too many unnecessary runs), as it might have some hoisting implications